src/Microsoft.AspNet.Mvc.Core/ActionContext.cs

```diff
@@ -56,23 +72,52 @@ public ActionContext([NotNull] ActionContext actionContext)
         }
 
         /// <summary>
-        /// Gets the <see cref="Mvc.ActionDescriptor"/> for the selected action.
+        /// Gets or sets the <see cref="Mvc.ActionDescriptor"/> for the selected action.
         /// </summary>
-        public ActionDescriptor ActionDescriptor { get; }
+        /// <remarks>
+        /// The setter should be used only when <see cref="ActionContext"/> is
+        /// directly instantiated in user codes, such as for unit tests.
```

Comment:
Again, too many words and it ends up being less clear.

`The property setter is provided for unit test purposes only.`